ff_derive/src/lib.rs

```diff
@@ -577,10 +590,11 @@ fn prime_field_constants_and_sqrt(
         };
 
     // Compute R^2 mod m
-    let r2 = biguint_to_u64_vec((&r * &r) % modulus, limbs);
+    let r2 = biguint_to_u64_vec(to_mont(r.clone()), limbs);
```

Comment:
Hmm. Yes I guess this is the same computation, but it's a bit odd. I think it might be more obvious to revert this change.

Reply:
Ah yes, this was caught in an automated refactor. I agree this specific case should be reverted.